tests/integration/e2e/MultiYearPlan.test.tsx

```diff
@@ -0,0 +1,183 @@
+        </MemoryRouter>
+      );
+    });
+    context('when a course is a child of another', function () {
+      it('should display the faculty of its parent course', async function () {
+        await renderResult.findByText(childCourse.title, { exact: false });
+        // Edit the child course to set its parent via the sameAs field
+        const editButton = await renderResult
+          .findByLabelText(`Edit course information for ${childCourse.title}`,
+            { exact: false });
+        fireEvent.click(editButton);
+        await renderResult.findByRole('dialog');
+        const sameAsSelect = renderResult.getByLabelText('Same As', { exact: false }) as HTMLSelectElement;
+        fireEvent.change(
+          sameAsSelect,
+          { target: { value: parentCourse.id } }
+        );
+        const submitButton = renderResult.getByText('Submit');
+        fireEvent.click(submitButton);
+        await wait(() => !renderResult.queryByText('required fields', { exact: false }));
+
+        // Navigate to multi year plan tab
+        const multiYearPlanTab = await renderResult.findByText('4 Year Plan', { exact: false });
+        fireEvent.click(multiYearPlanTab);
+        await renderResult.findByText(childCourse.title, { exact: false });
+
+        // Find the instructor information across the years for the parent course
+        const rows = renderResult.getAllByRole('row');
+        const utils = within(rows[1]);
+        const catalogNumberField = utils.queryByLabelText('The table will be filtered as characters are typed in this catalog number filter field');
+        fireEvent.change(catalogNumberField, { target: { value: `${parentCourse.prefix} ${parentCourse.number}` } });
+        const parentRow = Array.from(renderResult.getAllByRole('row'))[2] as HTMLTableRowElement;
+        let parentRowContent = (Array.from(parentRow.cells)
+          .map((cell) => cell.textContent));
+        // Remove the course top level information and
+        // take only the instructor information.
+        parentRowContent = parentRowContent.slice(3);
+
+        // Find the instructor information across the years for the child course
+        fireEvent.change(catalogNumberField, { target: { value: `${childCourse.prefix} ${childCourse.number}` } });
+        const childRow = Array.from(renderResult.getAllByRole('row'))[2] as HTMLTableRowElement;
+        let childRowContent = (Array.from(childRow.cells)
+          .map((cell) => cell.textContent));
+        // Remove the top level course information and
+        // take only the instructor information.
+        childRowContent = childRowContent.slice(3);
+
+        // Compare the faculty information and make sure they are equal
+        let areRowsEqual = true;
+        for (let i = 0; i < parentRowContent.length; i++) {
+          if (parentRowContent[i] !== childRowContent[i]) {
+            areRowsEqual = false;
+          }
+        }
+        strictEqual(areRowsEqual, true);
```

Comment:
Ideally, we should compare values directly in the assertion, instead of computing and comparing booleans like this. Mainly, this is to give us a higher fidelity error message (i.e. `Expected "A, B, C", got "X, Y, Z"` instead of `Expected true, got false`).

In this case, I think we further want to ensure that the `parentRowContent` and `childRowContent` actually contain data -- otherwise, we could get a passing test with an empty array (or array of empty strings) without actually checking that the child is showing the parent data.

Reply:
That's a good point about the rows possibly not containing data.

I ended up trying to tackle this from a different direction by finding an instructor first and setting the parent course using the course associated with the instructor I found. I figured this way it would guarantee that the course I was selecting as the parent was guaranteed to be associated with at least one instructor.

I also added a counter variable to check the number of rows that had been compared to further make sure that we were not comparing empty arrays.

For some reason, the multi year plan tests is timing out in the build, but it is passing locally for me over many tries. Do you think there is something that was added that is causing a time out? If not, I could alternatively try separating out the `strictEqual` statements into different tests.

Reply:
The timeout issue was resolved in [d4f7eed](https://github.com/seas-computing/course-planner/pull/652/commits/d4f7eedb784951766c323b2d3d283ab09dc7a56f) by increasing the timeout limit to 120 seconds.